src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildASelfContainedApp.cs

```diff
@@ -404,6 +404,32 @@ public void It_does_not_build_SelfContained_due_to_PublishSelfContained_being_tr
             outputDirectory.Should().NotHaveFile("hostfxr.dll"); // This file will only appear if SelfContained. 
         }
 
+        [Theory]
+        [InlineData("--p:PublishReadyToRun=true")]
+        [InlineData("-p:PublishSingleFile=true")]
+        [InlineData("-p:PublishSelfContained=true")]
+        [InlineData("-p:PublishAot=true")]
+        public void It_builds_without_implicit_rid_with_RuntimeIdentifier_specific_during_publish_only_properties(string property)
+        {
+            var tfm = ToolsetInfo.CurrentTargetFramework;
+            var testProject = new TestProject()
+            {
+                IsExe = true,
+                TargetFrameworks = tfm,
+            };
+            testProject.RecordProperties("RuntimeIdentifier");
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: $"ItBuildsWithoutImplicitRIDOnRIDRequiringPropertiesDuringPublish_{property}");
```

Comment:
You shouldn't need to add a bunch of stuff to the identifier, it just needs to be unique within the test name.

```suggestion
            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: property);
```

Reply:
Oh!